src/games/strategy/triplea/delegate/MoveValidator.java

```diff
@@ -1427,11 +1427,13 @@ public static String validateCanal(final Route route, final Collection<Unit> uni
     final Set<CanalAttachment> canalAttachments = CanalAttachment.get(territory);
     for (final CanalAttachment canalAttachment : canalAttachments) {
       if (!isCanalOnRoute(canalAttachment, route, data)) {
-        continue;
+        continue; // Only check canals that are on the route
       }
       failureMessage = canPassThroughCanal(canalAttachment, units, player, data);
-      if (!failureMessage.isPresent()) {
-        return Optional.empty();
+      final boolean canPass = !failureMessage.isPresent();
+      if ((!Properties.getControlAllCanalsBetweenTerritoriesToPass(data) && canPass)
```

Comment:
Use an `xor` to simplify the if statement?
`!(Properties.getControlAllCanalsBetweenTerritoriesToPass(data) ^ canPass)`